sdk/core/azure-core-test/src/test/java/com/azure/core/test/TestProxyTests.java

```diff
@@ -15,25 +15,50 @@
 import com.azure.core.test.utils.HttpURLConnectionHttpClient;
 import com.azure.core.util.Context;
 import com.azure.core.util.UrlBuilder;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
 import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Tag;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
 
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
 import java.net.MalformedURLException;
 import java.net.URL;
+import java.nio.file.Files;
 import java.util.ArrayList;
+import java.util.LinkedList;
 import java.util.List;
+import java.util.Map;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
 @SuppressWarnings("deprecation")
 public class TestProxyTests extends TestBase {
     static TestProxyTestServer server;
+    private static final String LOCAL_FILE_PATH = "src/test/resources/session-records/";
+    private static final ObjectMapper RECORD_MAPPER = new ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);
 
     private static List<TestProxySanitizer> customSanitizer = new ArrayList<>();
+
+    public static final String REDACTED = "REDACTED";
+
     static {
-        customSanitizer.add(new TestProxySanitizer("$..modelId", "REDACTED", TestProxySanitizerType.BODY));
+        customSanitizer.add(new TestProxySanitizer("$..modelId", REDACTED, TestProxySanitizerType.BODY));
+    }
+
+    private TestInfo testInfo;
+
+    @BeforeEach
+    void init(TestInfo testInfo) {
+        this.testInfo = testInfo;
```

Comment:
I'm suspicious of this.. there is already a `@BeforeEach` on `TestBase`. What is the behavior when two of them exist?

Reply:
oh nevermind, I was thinking of the `@BeforeAll`.